tasks/rsync.py

```diff
@@ -0,0 +1,218 @@
+             Don't use this param during parallel execution. if used then finally block will be delayed
+
+    data_dir: Optional Param. Specify the client like client.0/client.1
+              make sure cephfs mount and workunit task using this client.
+              Default source directory will be source/subdir for other scenarios.
+
+    snapenable: Optional Param. value in boolean (True or False), default False(disabled).
+                using this option will enable cephfs snapshot and rsync.
+
+    waittime: Optional Param. units in seconds. default 5 sec.
+              During rsync iteration each loop will wait for specified seconds.
+
+    Note: snapenable and data_dir can't co-exist.
+
+    Examples:
+
+    rsync task parallel with another task
+    tasks:
+      - ceph:
+      - ceph-fuse:
+      - rsync:
+            waittime: 10
+            snapenable: True
+            filesize: 8
+      - cephfs_test_runner:
+
+    rsync task parallel with workunit
+    tasks:
+      - ceph:
+      - ceph-fuse: [client.2]
+      - rsync:
+            waittime: 10
+            data_dir: client.2
+      - workunit:
+            clients:
+            client.2:
+                - suites/iozone.sh
+
+    rsync task sequentially based on runtime. default runtime is 0
+    tasks:
+      - ceph:
+      - ceph-fuse:
+      - rsync:
+            runtime: 120
+
+'''
+
+class RSync(Greenlet):
+
+    def __init__(self, ctx, config, logger):
+        super(RSync, self).__init__()
+
+        self.ctx = ctx
+
+        self.stopping = Event()
+        self.config = config
+        self.logger = logger
+
+        self.my_mnt = None
+        self.workUnit = False
+
+        self.runTime = self.config.get('runtime', 0)
+        self.fileSize = self.config.get('filesize', 2)
+        self.waitTime = self.config.get('waittime', 5)
+
+        self.fs = Filesystem(self.ctx)
+
+        self.snapEnable = bool(self.config.get('snapenable', False))
+        self.logger.info('Snapenable param: {}'.format(self.snapEnable))
+
+        if self.config.get('data_dir'):
+            self.workUnit = True
+            self.dataDir = misc.get_testdir(self.ctx)
+            self.sourceDir = self.dataDir + '/workunit.{}/'.format(self.config.get('data_dir'))
+        else:
+            self.snapShot = None
+            self.snapDummy = None
+            self.iteration = 0
+            self.dataDir = 'source'
+            self.sourceDir = self.dataDir + '/subdir'
+
+    def _run(self):
+        try:
+            self.do_rsync()
+        except:
+            # Log exceptions here so we get the full backtrace (it's lost
+            # by the time someone does a .get() on this greenlet)
+            self.logger.exception("Exception in do_rsync:")
+            raise
+
+    def log(self, x):
+        """Write data to logger assigned to this MDThrasher"""
+        self.logger.info(x)
+
+    def stop(self):
+        time.sleep(self.runTime)
```

Comment:
This will be useful if the user executes rsync as sequential. for eg: if this is not set then rsync while loop iteration happens only once, so added wait time which will make sure rsync runs till runTime.